src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,54 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 1;
+    private final int[] array;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+    }
+
+    @Override
+    protected void compute() {
+        int start = 0;
+        int workLoad = array.length;
+
+        if (workLoad > THRESHOLD) {
+            int middle = workLoad / 2;
+
+            int[] first = Arrays.copyOfRange(array, start, middle);
+            int[] second = Arrays.copyOfRange(array, middle, workLoad);
+
+            MergeSortAction firstSort = new MergeSortAction(first);
+            MergeSortAction secondSort = new MergeSortAction(second);
+
+            invokeAll(firstSort, secondSort);
+            merge(first, second);
```

Comment:
The 'merge' method is called without any arguments to specify the starting position in the original array. This will lead to incorrect results when merging subarrays back into the original array. You need to pass the starting index to the 'merge' method.